Azure/sonic-buildimage
# Restart SwSS, syncd and dependent services if a critical process in syncd container exits unexpectedly

**- What I did**

Restart SwSS, syncd and dependent services if a critical process in the syncd container exits unexpectedly

**- How I did it**

Add the same mechanism I developed for the SwSS service in https://github.com/Azure/sonic-buildimage/pull/2845 to the syncd service. However, in order to cause the SwSS service to *also* exit and restart in this situation, I developed a `docker-wait-any` program which the SwSS service uses to wait for *either* the swss *or* syncd containers to exit.

**- How to verify it**

Run etiher `sudo pkill -11 <critical_process_in_syncd_container>`, and observe that syncd service exits, swss and all dependent services exit, then all of those services start back up.
